### ISSUE TITLE
terraform_workspace_remote rule fails with dynamic blocks

### ISSUE THREAD

**Issue by oasys** [terraform-linters/tflint#766]
with the new `terraform_workspace_remote` rule enabled, tflint fails parsing dynamic blocks with:

```
Failed to check `terraform_workspace_remote` rule. An error occurred:

Error: Invalid reference: A reference to a resource type must be followed by at least one attribute access, specifying the resource name.
```

Here's a simplified config that causes this failure.

```HCL
terraform {
  backend "remote" {
    hostname     = "app.terraform.io"
    organization = "myorg"
    workspaces {
      name = "myworkspace"
    }
  }
}

locals {
  subnets = {
    private1 = {
      id = "1234"
      ip = "1.1.1.1"
    }
  }
}

resource "aws_route53_resolver_endpoint" "in" {
  direction          = "INBOUND"
  security_group_ids = ["1"]
  dynamic "ip_address" {
    for_each = local.subnets
    iterator = subnet
    content {
      subnet_id = subnet.value.id
      ip        = subnet.value.ip
    }
  }
}
```

**Comment by oasys** [comment]
It looks like this was fixed by #759 
I tested with 0.16.1, and it parses correctly.
